File: app/lib/features/chat/domain/chat_session.dart
/// Represents a chat session with the AI.
class ChatSession {
  const ChatSession({
    required this.id,
    required this.userId,
    this.title,
    required this.category,
    required this.language,
    required this.messageCount,
    required this.createdAt,
    required this.updatedAt,
  });

  final String id;
  final String userId;
  final String? title;
  final String category;
  final String language;
  final int messageCount;
  final DateTime createdAt;
  final DateTime updatedAt;

  factory ChatSession.fromJson(Map<String, dynamic> json) {
    return ChatSession(
      id: json['id'] as String,
      userId: json['user_id'] as String,
      title: json['title'] as String?,
      category: (json['category'] as String?) ?? 'general',
      language: (json['language'] as String?) ?? 'en',
      messageCount: (json['message_count'] as int?) ?? 0,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }

  ChatSession copyWith({
    String? id,
    String? userId,
    String? title,
    String? category,
    String? language,
    int? messageCount,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return ChatSession(
      id: id ?? this.id,
      userId: userId ?? this.userId,
      title: title ?? this.title,
      category: category ?? this.category,
      language: language ?? this.language,
      messageCount: messageCount ?? this.messageCount,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: backend/ai_service/models/chat_session.py
"""ChatSession model — maps to chat_sessions table (DATA_MODEL.md §2)."""

import uuid
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column

from database import Base


class ChatSession(Base):
    """AI chat session."""

    __tablename__ = "chat_sessions"

    id: Mapped[str] = mapped_column(
        String(36), primary_key=True, default=lambda: str(uuid.uuid4())
    )
    user_id: Mapped[str] = mapped_column(
        String(128), ForeignKey("profiles.id"), nullable=False
    )
    title: Mapped[str | None] = mapped_column(
        String(200), nullable=True, default=None
    )
    category: Mapped[str] = mapped_column(
        String(30), nullable=False, default="general"
    )
    language: Mapped[str] = mapped_column(
        String(5), nullable=False, default="en"
    )
    message_count: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, server_default=func.now()
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, server_default=func.now(),
        onupdate=func.now()
    )
    deleted_at: Mapped[datetime | None] = mapped_column(
        DateTime(timezone=True), nullable=True, default=None
    )
